### PR TITLE
Improve the quality of fetched favicons

### PR DESCRIPTION
**Pull request by samsymons**
Task/Issue URL: https://app.asana.com/0/414235014887631/1198210503453245/f
Tech Design URL:
CC:

**Description**:

This PR improves the quality of fetched favicons. It does so by augmenting the existing favicon fetching code to use the `LinkPresentation` first, which is a framework designed to fetch metadata for URLs.

**Steps to test this PR**:
1. Run the app on develop and bookmark as many domains as you can think of. See the Asana task for a shortlist of domains which currently display in lower resolutions in the app.
1. Clear the domain cache (there's a new option in the debug menu for this) and load favicons again, checking that they increase in quality.

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPad

**OS Testing**:

* [ ] iOS 13
* [ ] iOS 14

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**

### CODE REVIEW

File: DuckDuckGo/Favicons.swift [hunk 1]
```diff
@@ -292,6 +304,26 @@ public class Favicons {
         }
     }
 
+    @available(iOS 13.0, *)
```
Comment (brindy) [inline DuckDuckGo/Favicons.swift]: Unneeded?

File: DuckDuckGo/Favicons.swift [hunk 2]
```diff
@@ -292,6 +304,26 @@ public class Favicons {
...
+    private func retrieveLinkPresentationImage(from domain: String, completion: @escaping (UIImage?) -> Void) {
+        guard let url = URL(string: "http://\(domain)") else {
```
Comment (brindy) [inline DuckDuckGo/Favicons.swift]: Should we stick to https? http is handled in the fallback already, better to have a bad icon than potentially leak over http.

File: DuckDuckGo/ImageCacheDebugViewController.swift [hunk 3]
```diff
@@ -18,6 +18,7 @@
 //
 
 import UIKit
+import Core
```
Comment (brindy) [inline DuckDuckGo/ImageCacheDebugViewController.swift:21]: Unneeded version check on this class

**Review by brindy — APPROVED** [review]
Few minor things, but otherwise looks good. :)

File: DuckDuckGo/ImageCacheDebugViewController.swift (hunk 3 above)
Comment (samsymons) [reply DuckDuckGo/ImageCacheDebugViewController.swift:21]: Thanks, fixed - this shows how old this branch actually is. 🙂

File: DuckDuckGo/Favicons.swift (hunk 2 above)
Comment (samsymons) [reply DuckDuckGo/Favicons.swift]: Yeah good call. I made this change and tested it on a number of cases, didn't notice any impact on favicon quality, so I've pushed the HTTPS change up. :+1:

**Pull request merged**